### DISCUSSION TITLE
Group level role based authentication

### DISCUSSION THREAD

**Discussion by lukvermeulen** [nextauthjs/next-auth#1780] [Help]
Thanks for this awesome project!

**Your question**
What would be the best way to achieve group level role based authentication? 

**What are you trying to do**
Imagine I have multiple communities, and in those communities I have multiple user roles (Admin, User, ..). So one User could have Admin rights in one community, but have no rights in other communities.

**Feedback**
I found several Issues that give pointers on how to achieve role based authentication, but in a global scope. If anyone has pointers on how to achieve this in group level, that would be much appreciated! Also best practices on how to set this up in general would be welcome. Thanks!

* [ ] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [x] Could not find relevant documentation
* [ ] Found the example project helpful
* [x] Did not find the example project helpful

**Comment by StefanSelfTaught** [comment]
Does [this](https://github.com/nextauthjs/next-auth/discussions/805) help?

**Reply by lukvermeulen** [reply]
Thank you for your answer StefanSelfTaught! I had already seen this issue, but concluded for myself that this would generate global user roles, and not on group level. I thought of implementing group level roles in the same way, but prefixing the role with the name of the community e.g. MY_COMMUNITY.ROLE_ADMIN. I suspected that that would be a "hacky" way, or am I mistaken here and that method would be fine to use? Thanks again!
